config/registry/aws/modules/aws-s3.md

```diff
@@ -24,6 +24,34 @@ This module creates an S3 bucket for storage purposes. It is created with server
         - "runx.dev"
 ```
 
+### File uploading
+To upload files to s3, just set the `files` field to the path (relative to the yaml or absolute) of a local directory. 
+On the next apply, all the files and subdirectories will be automatically uploaded to the bucket!
+
+
+So for example if you called the module like so:
+```yaml
+  - type: s3
+    name: blah
+    bucket_name: "opta-is-testing-cloudfront"
+    files: "../blah"
+```
+
+And the `../blah` directory had the following structure:
+```
+../blah
+├── hello2.html
+├── hello2.txt
+├── hello3.txt
+└── subdir
+    └── hello3.html
+```
+
+Then Opta would upload 4 files to your S3 bucket, with the S3 keys being `hello2.html`, `hello2.txt`, `hello3.txt` and
+`subdir/hello3.html`.
+
+Opta will also catch any changes to the files on the next `opta apply` and will push updates as needed. Opta supports
+extensive MIME parsing, so it also makes sure to set the content type correctly.
 
 ### Linking
```

Comment:
We should add here that this module can be linked to cloudfront to serve static sites (link to cloudfront docs)

Also we should document on this page somewhere that we are creating a cloudfront OAI.

Reply:
done